legal-api/src/legal_api/resources/v2/configuration.py

```diff
@@ -35,3 +35,69 @@ def get_configurations():
             configuration.json for configuration in configurations
         ]
     }), HTTPStatus.OK
+
+
+@bp.route('', methods=['PUT'])
+@cross_origin(origin='*')
+@jwt.has_one_of_roles([UserRoles.staff])
+def update_configurations():
+    """Update the configurations."""
+    json_input = request.get_json()
+    if not json_input:
+        return {'message': 'Request body cannot be blank'}, HTTPStatus.BAD_REQUEST
+
+    configurations = json_input.get('configurations', [])
+    if not configurations:
+        return {'message': 'Configurations list cannot be empty'}, HTTPStatus.BAD_REQUEST
+
+    valid_names, msg = has_validate_names(configurations)
+    if not valid_names:
+        return ({'message': msg}), HTTPStatus.BAD_REQUEST
+
+    numeric_names = {'NUM_DISSOLUTIONS_ALLOWED', 'MAX_DISSOLUTIONS_ALLOWED'}
+
+    for config_data in configurations:
+        name = config_data.get('name')
+        value = config_data.get('value')
+
+        if name in numeric_names:
+            if not is_valid_numeric_value(name, value):
+                return {'message': f'Invalid value for {name}.'}, HTTPStatus.BAD_REQUEST
+
+        config = Configuration.find_by_name(name)
+        config.val = str(value)
+        try:
+            config.save()
```

Comment:
As per our conversation, it would be ideal if we only save entries only add the end. 

If there are any validation errors, we would roll back all updates to updated configuration.

Reply:
Updated